### DISCUSSION TITLE
Show battery voltage or percentage remaining on battery charge

### DISCUSSION THREAD

**Discussion by daveclifford** [DCC-EX/EX-T3-WiFi#83] [Ideas]
It may be nice to have the Throttle show estimated battery time remaining or voltage (or both) when turned on - maybe for few seconds - so the absent-minded are told to re-charge the throttle.

You would have to know the battery capacity - mwh

Not a high priority feature  - consider it a nice to have!

Dave

**Comment by alex-code** [comment]
I did hope to have a battery percentage but I can only access the current voltage and from reading LiPo batteries have a non linear voltage curve so you can't get a estimate from voltage alone.

The best I could do was the 3 battery icons.

Adafruit does a small board which sits between the LiPo and the ESP32 which can give more detail but it adds more complexity.
This is really a feature for a future custom board 😁

**Reply by daveclifford** [reply]
Nice response from you - there are several articles on the web describing how-tos

[https://how2electronics.com/iot-based-battery-status-monitoring-system-using-esp8266/]

Your current implementation is fine.  An operating session should be 2-3 hours, so the large battery charged should suffice.

Dave

**Reply by alex-code** [reply]
I'd looked at this https://www.adafruit.com/product/4712 but it would make the wiring more complex.

**Comment by daveclifford** [comment]
I've always liked the KISS principle.

Dave

**Reply by Alan-Lomax** [reply]
As a future idea ... In order to wring the most out of what we have (especially if an accelerometer is on board) .. could we not put the throttle to sleep after a minutes or two and then wake it up again on detecting motion?.     I was reading further about the 3 CPU's in an ESP32 and how one is an ultra low power one that can still wake up the other CPU's on pin change. The other two CPU's are of course the main 32 bit cores.

**Reply by alex-code** [reply]
This was something I looked into but unfortunately due to the way the TFT display is made if I put it to sleep I can't wake it up.

I did think about turning off the display backlight after inactivity but then it would be easy to forget it was still turned on.

**Comment by daveclifford** [comment]
closed
